utils/swift_build_support/swift_build_support/products/cmake_product.py

```diff
@@ -23,6 +23,10 @@ class CMakeProduct(product.Product):
     def is_verbose(self):
         return self.args.verbose_build
 
+    @property
+    def cmake_os_sysroot(self):
```

Comment:
Why the property function instead of setting `self.cmake_os_sysroot`? If we get an error for trying to use `cmake_os_sysroot` before calling `host_cmake_options`, this will still result in a failure because `self._cmake_os_sysroot` is still undefined so it isn't protecting us from anything.

Reply:
I wanted to make it discoverable that this value can now be accessed outside of this class, and I thought that adding a read only property would help convey this (instead of trawling through the code of `build_with_cmake`)

Reply:
I reworked my approach to avoid the use of the property, and instead return if from `host_cmake_options`